### ISSUE TITLE
[HTTP] Reply hangs when empty payload is sent

### ISSUE THREAD

**Issue by yasassri** [ballerinalang/ballerina#3350]
**Steps to Reproduce**

1. Create the following Bal service.
```
package resources.services;

import ballerina.net.http;
import ballerina.lang.messages;

@http:configuration {basePath:"/encoded"}
service<http> EncodingService {

    @http:resourceConfig {
        methods:["POST","GET","PUT","PATCH"],
        path:"/spayload"
    }
    resource s_echo (message m) {
            string payload = messages:getStringPayload(m);
	    http:convertToResponse(m);
            reply m;
    }
}

```
2. Invoke the service with a Post request without a payload.

**Issue:**  Ballerina hangs and sends an internal server error after some time. Following error was observed in the logs.

```
ballerina.lang.errors:Error, message: unknown error occurred
	at resources.services:EncodingService.s_echo(resources/services/EncodingService.bal:19)
```

**Comment by irunika** [comment]
This issue is fixed in v0.94

**Comment by shafreenAnfar** [comment]
As per the above comment closing the issue.
